snap/quota/resources_builder.go

```diff
@@ -70,6 +79,24 @@ func (rb *ResourcesBuilder) WithThreadLimit(limit int) *ResourcesBuilder {
 	return rb
 }
 
+func (rb *ResourcesBuilder) WithJournalNamespace() *ResourcesBuilder {
```

Comment:
I didn't know if it would be ever neccessary, I just wanted to add it initially here in case we would ever want such a functionality. It didn't seem far-fetched to me people want just isolate their snaps in a different namespace. I can remove this support easily if you feel like its not neccessary to have this. We can also leave the support in and not expose it.